spec/lib/ulid_spec.rb

```diff
@@ -68,4 +68,31 @@
       end
     end
   end
+
+  describe 'decoding a timestamp' do
+    it 'decodes the timestamp as Time' do
+      ulid = ULID.generate
+      assert ULID.decode_time(ulid).is_a?(Time)
+    end
+
+    it 'decodes the timestamp into milliseconds' do
+      input_time = Time.now.utc
+      ulid = ULID.generate(input_time)
+      output_time = ULID.decode_time(ulid).utc
+      assert_equal (input_time.to_f * 1000).to_i, (output_time.to_f * 1000).to_i
```

Comment:
@andjosh this test is flaky - it passes most of the time, but it does fail sometimes.

```
  1) Failure:
ULID::decoding a timestamp#test_0002_decodes the timestamp into milliseconds [/Users/raf/code/ws/ulid/spec/lib/ulid_spec.rb:82]:
Expected: 1677986614361
  Actual: 1677986614360
```

Reply:
I reverted the merge for that reason. It seems that the decoder isn't extracting the correct precision